src/epd2in13_v3/command.rs

```diff
@@ -0,0 +1,276 @@
+}
+
+#[allow(dead_code)]
+#[derive(Copy, Clone)]
+pub(crate) enum BorderWaveFormGs {
+    Lut0 = 0x0,
+    Lut1 = 0x1,
+    Lut2 = 0x2,
+    Lut3 = 0x3,
+}
+
+pub(crate) struct BorderWaveForm {
+    pub vbd: BorderWaveFormVbd,
+    pub fix_level: BorderWaveFormFixLevel,
+    pub gs_trans: BorderWaveFormGs,
+}
+
+impl BorderWaveForm {
+    pub fn to_u8(&self) -> u8 {
+        *0u8.set_bits(6..8, self.vbd as u8)
+            .set_bits(4..6, self.fix_level as u8)
+            .set_bits(0..2, self.gs_trans as u8)
+    }
+}
+
+#[allow(dead_code)]
+#[derive(Copy, Clone)]
+pub enum DeepSleepMode {
+    // Sleeps and keeps access to RAM and controller
+    Normal = 0x00,
+
+    // Sleeps without access to RAM/controller but keeps RAM content
+    Mode1 = 0x01,
+
+    // Same as MODE_1 but RAM content is not kept
+    Mode2 = 0x11,
+}
+
+pub(crate) struct GateDrivingVoltage(pub u8);
+pub(crate) struct SourceDrivingVoltage(pub u8);
+pub(crate) struct Vcom(pub u8);
+
+pub(crate) trait I32Ext {
+    fn vcom(self) -> Vcom;
+    fn gate_driving_decivolt(self) -> GateDrivingVoltage;
+    fn source_driving_decivolt(self) -> SourceDrivingVoltage;
+}
+
+impl I32Ext for i32 {
+    // This is really not very nice. Until I find something better, this will be
+    // a placeholder.
+    fn vcom(self) -> Vcom {
+        assert!((-30..=-2).contains(&self));
+        let u = match -self {
+            2 => 0x08,
+            3 => 0x0B,
+            4 => 0x10,
+            5 => 0x14,
+            6 => 0x17,
+            7 => 0x1B,
+            8 => 0x20,
+            9 => 0x24,
+            10 => 0x28,
+            11 => 0x2C,
+            12 => 0x2F,
+            13 => 0x34,
+            14 => 0x37,
+            15 => 0x3C,
+            16 => 0x40,
+            17 => 0x44,
+            18 => 0x48,
+            19 => 0x4B,
+            20 => 0x50,
+            21 => 0x54,
+            22 => 0x58,
+            23 => 0x5B,
+            24 => 0x5F,
+            25 => 0x64,
+            26 => 0x68,
+            27 => 0x6C,
+            28 => 0x6F,
+            29 => 0x73,
+            30 => 0x78,
+            _ => 0,
+        };
+        Vcom(u)
+    }
+
+    fn gate_driving_decivolt(self) -> GateDrivingVoltage {
+        assert!((100..=210).contains(&self) && self % 5 == 0);
+        GateDrivingVoltage(((self - 100) / 5 + 0x03) as u8)
+    }
+
+    fn source_driving_decivolt(self) -> SourceDrivingVoltage {
+        assert!((24..=88).contains(&self) || (self % 5 == 0 && (90..=180).contains(&self.abs())));
+
+        if (24..=88).contains(&self) {
+            SourceDrivingVoltage(((self - 24) + 0x8E) as u8)
+        } else if (90..=180).contains(&self) {
+            SourceDrivingVoltage(((self - 90) / 2 + 0x23) as u8)
+        } else {
+            SourceDrivingVoltage((((-self - 90) / 5) * 2 + 0x1A) as u8)
+        }
+    }
+}
+
+impl traits::Command for Command {
+    /// Returns the address of the command
+    fn address(self) -> u8 {
+        self as u8
+    }
+}
```

Comment:
I haven't checked it yet, but if v2 and v3 only differ in the lut constants we could fully reuse this here instead